src/compas_timber/fabrication/text.py

```diff
@@ -0,0 +1,285 @@
+        else:
+            raise ValueError("Angle should be between -180 and 180. Got: {}".format(value))
+
+    @property
+    def alignment_vertical(self):
+        return self._alignment_vertical
+
+    @alignment_vertical.setter
+    def alignment_vertical(self, value):
+        if value not in ["bottom", "center", "top"]:
+            raise ValueError("AlignmentVertical should be one of 'bottom', 'center', 'top'. Got: {}".format(value))
+        self._alignment_vertical = value
+
+    @property
+    def alignment_horizontal(self):
+        return self._alignment_horizontal
+
+    @alignment_horizontal.setter
+    def alignment_horizontal(self, value):
+        if value not in ["left", "center", "right"]:
+            raise ValueError("AlignmentHorizontal should be one of 'left', 'center', 'right'. Got: {}".format(value))
+        self._alignment_horizontal = value
+
+    @property
+    def alignment_multiline(self):
+        return self._alignment_multiline
+
+    @alignment_multiline.setter
+    def alignment_multiline(self, value):
+        if value not in ["left", "center", "right"]:
+            raise ValueError("AlignmentMultiline should be one of 'left', 'center', 'right'. Got: {}".format(value))
+        self._alignment_multiline = value
+
+    @property
+    def stacked_marking(self):
+        return self._stacked_marking
+
+    @stacked_marking.setter
+    def stacked_marking(self, value):
+        if value not in [True, False]:
+            raise ValueError("StackedMarking should be a Boolean. Got: {}".format(value))
+        self._stacked_marking = value
+
+    @property
+    def text_height_auto(self):
+        return self._text_height_auto
+
+    @text_height_auto.setter
+    def text_height_auto(self, value):
+        if value not in [True, False]:
+            raise ValueError("TextHeightAuto should be a Boolean. Got: {}".format(value))
+        self._text_height_auto = value
+
+    @property
+    def text_height(self):
+        return self._text_height
+
+    @text_height.setter
+    def text_height(self, value):
+        if 0 <= value <= 50000:
+            self._text_height = value
+        else:
+            raise ValueError("TextHeight should be between 0 and 50000. Got: {}".format(value))
+
+    @property
+    def text(self):
+        return self._text
+
+    @text.setter
+    def text(self, value):
+        self._text = value
+
+
+    ########################################################################
+    # Methods
+    ########################################################################
+
+    def apply(self, geometry, beam):
+        """Apply the feature to the beam geometry.
+
+        Raises
+        -------
+
+        Returns
+        -------
+        :class:`compas.geometry.Brep`
+            The resulting geometry after processing. #TODO: think about ways to display text curves from `draw_string_on_element()`
+
+        """
+        return geometry
+
+    def draw_string_on_element(self, element):
```

Comment:
I understand we're not doing this at the moment but we might want to in the future, right?
could you add a docstring so it's clearer what this does?